mmv1/products/bigquerydatapolicy/api.yaml

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Product
+name: BigqueryDatapolicy
+display_name: BigQuery Data Policy
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://bigquerydatapolicy.googleapis.com/v1beta1/
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: BigQuery Data Policy API
+    url: https://console.cloud.google.com/apis/library/bigquerydatapolicy.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: "DataPolicy"
+    min_version: beta
+    base_url: projects/{{project}}/locations/{{location}}/dataPolicies
+    create_url: projects/{{project}}/locations/{{location}}/dataPolicies
+    self_link: projects/{{project}}/locations/{{location}}/dataPolicies/{{data_policy_id}}
+    update_verb: :PATCH
+    update_mask: true
+    description: A BigQuery Data Policy
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        "Official Documentation": "https://cloud.google.com/bigquery/docs/column-data-masking-intro"
+      api: "https://cloud.google.com/bigquery/docs/reference/bigquerydatapolicy/rest/v1beta1/projects.locations.dataPolicies/create"
+    iam_policy: !ruby/object:Api::Resource::IamPolicy
+      exclude: false
+      method_name_separator: ":"
+      fetch_iam_policy_verb: :POST
+      parent_resource_attribute: "data_policy_id"
+      import_format:
```

Comment:
Specifying this as the import format implies that we expect users to only specify the data_policy_id using the fully-qualified data policy's ID. 

I think we want to include `{{data_policy_id}}` here, to let users only specify the last part of the id. Alternatively we can change uses to all specify the `.id` field of the parent resource

Reply:
arent we going back and forth on this?
https://github.com/GoogleCloudPlatform/magic-modules/pull/6425#discussion_r956252910

Reply:
Yeah it looks like it! I want to make sure that all of the relevant tests pass though.

There are two options to fix this as I noted in the above comment

Reply:
You should be able to include {{location}}/{{data_policy_id}} in the import format, like the bigqueryconnection resource: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/bigqueryconnection

Reply:
It might also require adding that to the import ID of the actual data policy resource as well

Reply:
aligned with connection resource and ran tests